### PR TITLE
Fragile mind nloths gift

### PR DESCRIPTION
**Pull request by wang429**
1. Fragile Mind can't be offered to N'loth. Tested by entering the event with any other relics (total 3) multiple times using `event N'loth`.
2. N'loth event can't occur if there are only two relics, Fragile Mind and any other. Unsure how to test.

### CODE REVIEW

File: src/main/java/stsjorbsmod/patches/NlothsGiftPatch.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    }
+
+    @SpirePatch(clz = AbstractDungeon.class, method = "getShrine")
+    public static class AbstractDungeon_getShrine_NlothCheck {
+        public static ExprEditor Instrument() {
+            return new ExprEditor() {
+                @Override
+                public void edit(FieldAccess fieldAccess) throws CannotCompileException {
+                    if (fieldAccess.getClassName().equals(AbstractPlayer.class.getName())
+                            && fieldAccess.getFieldName().equals("relic")) {
```
Comment (dbjorge) [inline src/main/java/stsjorbsmod/patches/NlothsGiftPatch.java]: I think this should be `relics` plural

File: src/main/java/stsjorbsmod/patches/NlothsGiftPatch.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        removeFragileMindRelic(relics.iterator());
+        return relics;
+    }
+
+    @SpirePatch(clz = AbstractDungeon.class, method = "getShrine")
+    public static class AbstractDungeon_getShrine_NlothCheck {
+        public static ExprEditor Instrument() {
+            return new ExprEditor() {
+                @Override
+                public void edit(FieldAccess fieldAccess) throws CannotCompileException {
```
Comment (dbjorge) [inline src/main/java/stsjorbsmod/patches/NlothsGiftPatch.java]: For testing it, I think a good option would would be to add a second temporary path to getShrine that uses `JorbsMod.logger` to print out the `tmp` list after it's generated.

File: src/main/java/stsjorbsmod/patches/NlothsGiftPatch.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+     * @param playerRelics
+     * @return
+     */
+    public static List<AbstractRelic> clonePlayerRelicsWithoutFragileMind(List<AbstractRelic> playerRelics) {
+        List<AbstractRelic> relics = new ArrayList<>(playerRelics);
+        removeFragileMindRelic(relics.iterator());
+        return relics;
+    }
+
+    @SpirePatch(clz = AbstractDungeon.class, method = "getShrine")
```
Comment (dbjorge) [inline src/main/java/stsjorbsmod/patches/NlothsGiftPatch.java]: I think you could actually reuse exactly this same patch on `Nloth.class`/`SpirePatch.CONSTRUCTOR`, if you wanted to gain confidence that the patch implementation worked in the case that's easier to test. You can apply multiple `@SpirePatch` annotations to the same class.

File: src/main/java/stsjorbsmod/patches/NlothsGiftPatch.java [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    public static class AbstractDungeon_getShrine_NlothCheck {
+        public static ExprEditor Instrument() {
+            return new ExprEditor() {
+                @Override
+                public void edit(FieldAccess fieldAccess) throws CannotCompileException {
+                    if (fieldAccess.getClassName().equals(AbstractPlayer.class.getName())
+                            && fieldAccess.getFieldName().equals("relic")) {
+                        fieldAccess.replace("{ $_ = " + NlothsGiftPatch.class.getName() + ".clonePlayerRelicsWithoutFragileMind($0)); }");
```
Comment (dbjorge) [inline src/main/java/stsjorbsmod/patches/NlothsGiftPatch.java]: `$0` isn't what you want here; when you're doing a field access patch for original code `someObject.someField`, `$0` represents `someObject`. What you're going to want to do is going to look almost identical to the `CloneMasterDeckWithoutLegendaryCardsEditor` class in `LegendaryPatch`; I'd recommend taking a look at what that's doing :)

File: src/main/java/stsjorbsmod/patches/NlothsGiftPatch.java (hunk 3 above)
Comment (dbjorge) [reply src/main/java/stsjorbsmod/patches/NlothsGiftPatch.java]: (or alternately, have separate @SpirePatch classes but have them both be based on a shared `ExprEditor` class, like how `LegendaryPatch.java` reuses `CloneMasterDeckWithoutLegendearyCardsEditor` for a few different `@SpirePatch`es)

File: src/main/java/stsjorbsmod/patches/NlothsGiftPatch.java (hunk 1 above)
Comment (wang429) [reply src/main/java/stsjorbsmod/patches/NlothsGiftPatch.java]: facepalming intensifies.

File: src/main/java/stsjorbsmod/patches/NlothsGiftPatch.java (hunk 3 above)
Comment (wang429) [reply src/main/java/stsjorbsmod/patches/NlothsGiftPatch.java]: Hmm I'll try this.

File: src/main/java/stsjorbsmod/patches/NlothsGiftPatch.java (hunk 2 above)
Comment (wang429) [reply src/main/java/stsjorbsmod/patches/NlothsGiftPatch.java]:
```
02:07:32.293 INFO stsjorbsmod.JorbsMod> [Match and Keep!, Wheel of Change, Golden Shrine, Transmorgrifier, Upgrade Shrine, Accursed Blacksmith, Designer, Duplicator, FaceTrader, Fountain of Cleansing, Lab, NoteForYourself, The Joust, The Woman in Blue]
```
output from that line with only the starting relics. N'loth is not there.

**Pull request merged**